### DISCUSSION TITLE
Want to use  --theme dark --always-high-vram --attention-split

### DISCUSSION THREAD

**Discussion by kengun2077** [runew0lf/RuinedFooocus#150] [General]
Want to use
entry_with_update.py --theme dark --always-high-vram --attention-split

**Comment by runew0lf** [comment]
good for you, this isnt fooocus or auto1111

**Reply by kengun2077** [reply]
Dear, after trying and failing again and again, RuinedFooocus can use Colab HIGH_VRAM when adding some code to the files webui.py and launch.py​​

!python /content/RuinedFooocus/entry_with_update.py  --highvram --share --cuda-malloc --cuda-device 0

------------------------------

The file  webui.py add some code

import comfy.model_management as model_management
from comfy.cli_args import args


def get_parser():
    parser = argparse.ArgumentParser()
    parser.add_argument("--cpu", action="store_true", help="To use the CPU for everything (slow).")
    parser.add_argument("--highvram", action="store_true")
    parser.add_argument("--use-pytorch-cross-attention", action="store_true", help="Use the new pytorch 2.0 cross attention function.")
    parser.add_argument("--cuda-malloc", action="store_true", help="Enable cudaMallocAsync (enabled by default for torch 2.0 and up).")
    parser.add_argument("--cuda-device", type=int, default=None, metavar="DEVICE_ID", help="Set the id of the cuda device this instance will use.")
    parser.add_argument("--port", type=int, default=None, help="Set the listen port.")


-------------------------------

The file  launch.py add some code

import argparse
import gradio as gr


def clear_comfy_args():
    argv = sys.argv
    sys.argv = [sys.argv[0]]
    import comfy.cli_args
    sys.argv = argv
    parser = argparse.ArgumentParser('launch.py', parents=[comfy.cli_args.parser], conflict_handler='resolve')
    parser.add_argument("--cpu", action="store_true", help="To use the CPU for everything (slow).")
    parser.add_argument("--share", action='store_true', help="Set whether to share on Gradio.")
    parser.add_argument("--highvram", action="store_true")
    parser.add_argument("--use-pytorch-cross-attention", action="store_true", help="Use the new pytorch 2.0 cross attention function.")
    parser.add_argument("--port", type=int, default=None, help="Set the listen port.")
    parser.add_argument("--share", action='store_true', help="Set whether to share on Gradio.")
    parser.add_argument("--listen", type=str, default=None, metavar="IP", nargs="?", const="0.0.0.0", help="Set the listen interface.")
    parser.add_argument("--cuda-malloc", action="store_true", help="Enable cudaMallocAsync (enabled by default for torch 2.0 and up).")
    parser.add_argument("--cuda-device", type=int, default=None, metavar="DEVICE_ID", help="Set the id of the cuda device this instance will use.")
    comfy.cli_args.args = parser.parse_args()

**Reply by kengun2077** [reply]
Dear, after trying and failing again and again, RuinedFooocus change color ,
modify webui.py

---------------
if settings["theme"] == "None":
    theme = gr.themes.Glass()
else:
    theme = settings["theme"]

-------------------

if settings["theme"] == "None":
    theme = gr.themes.Monochrome()
else:
    theme = settings["theme"]

--------------------------


if settings["theme"] == "None":
    theme = 'JohnSmith9982/small_and_pretty'
else:
    theme = settings["theme"]

-----------------------------
![未命名](https://github.com/user-attachments/assets/2e596a8e-6e9a-469a-91f9-71d8a3dd66c7)

------------------------------
https://huggingface.co/spaces/JohnSmith9982/small_and_pretty

https://huggingface.co/spaces/gradio/theme-gallery
----------------------------


Gradio comes with a set of pre-built themes:
gr.themes.Base()
gr.themes.Default()
gr.themes.Glass()
gr.themes.Monochrome()
gr.themes.Soft()

https://www.gradio.app/guides/theming-guide

**Reply by kengun2077** [reply]
Dear, after trying and failing again and again, 
RuinedFooocus interface can be changed to dark 

Need 2 steps

Step 1.
chrome://flags
Enabled [ Auto Dark Mode for Web Contents ]
Relaunch Chrome

![1](https://github.com/user-attachments/assets/f7f657a8-5351-4584-8bde-30bcf2f3c428)

Step 2.
add    /?__theme=dark

https://e5f1ba177219c1dc41.gradio.live/?__theme=dark

http://127.0.0.1:7860/?__theme=dark
![2](https://github.com/user-attachments/assets/57a9fa38-2175-4859-81a4-76dad39b9dc5)

**Reply by kengun2077** [reply]
RuinedFooocus interface can be changed to dark  method 2


Step 1.
chrome://settings/appearance
Mode dark

![未命名](https://github.com/user-attachments/assets/cb0494bf-091c-439c-9864-84a155db2cd2)

Step 2.
add /?__theme=dark

https://e5f1ba177219c1dc41.gradio.live/?__theme=dark
